src/Core/WinSWCore/Util/XmlHelper.cs

```diff
@@ -87,22 +87,33 @@ public static TAttributeType SingleAttribute<TAttributeType>(XmlElement node, st
         /// <returns>Attribute value (or default)</returns>
         /// <exception cref="InvalidDataException">Wrong enum value</exception>
         public static TAttributeType EnumAttribute<TAttributeType>(XmlElement node, string attributeName, TAttributeType defaultValue)
+            where TAttributeType : struct
         {
             if (!node.HasAttribute(attributeName))
                 return defaultValue;
 
             string rawValue = node.GetAttribute(attributeName);
             string substitutedValue = Environment.ExpandEnvironmentVariables(rawValue);
+#if NET20
             try
             {
                 var value = Enum.Parse(typeof(TAttributeType), substitutedValue, true);
                 return (TAttributeType)value;
             }
-            catch (Exception ex) // Most likely ArgumentException
+            catch (ArgumentException ex)
```

Comment:
I have no memory of that place

Reply:
There's only `ArgumentException`:

https://docs.microsoft.com/dotnet/api/system.enum.parse#System_Enum_Parse_System_Type_System_String_System_Boolean_